src/test/java/lambda/part1/exercise/Exercise1.java

```diff
@@ -0,0 +1,126 @@
+package lambda.part1.exercise;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
+import java.util.Comparator;
+import lambda.data.Person;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+public class Exercise1 {
+
+    private Comparator<Person> personComparator = new Comparator<Person>() {
+        @Override
+        public int compare(Person o1, Person o2) {
+            return Integer.compare(o1.getAge(), o2.getAge());
+        }
+    };
+
+    private Comparator<Person> personByLastAndFirstNamesComparator = new Comparator<Person>() {
+        @Override
+        public int compare(Person o1, Person o2) {
+            int lastNameCompare = o1.getLastName().compareTo(o2.getLastName());
+            if (lastNameCompare != 0) {
+                return lastNameCompare;
+            }
+            return o1.getFirstName().compareTo(o2.getFirstName());
+        }
+    };
+
+    @Test
+    public void sortPersonsByAgeUsingArraysSortComparator() {
+        Person[] persons = getPersons();
+
+        Arrays.sort(persons, new Comparator<Person>() {
+            @Override
+            public int compare(Person o1, Person o2) {
+                return Integer.compare(o1.getAge(), o2.getAge());
+            }
+        });
+
+        assertArrayEquals(new Person[]{
+            new Person("Иван", "Мельников", 20),
+            new Person("Николай", "Зимов", 30),
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45)
+        }, persons);
+    }
+
+    @Test
+    public void sortPersonsByAgeUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        Arrays.sort(persons,personComparator);
+
+        assertArrayEquals(new Person[]{
+            new Person("Иван", "Мельников", 20),
+            new Person("Николай", "Зимов", 30),
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45)
+        }, persons);
+    }
+
+    @Test
+    public void sortPersonsByLastNameThenFirstNameUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        Arrays.sort(persons, personByLastAndFirstNamesComparator);
+
+        assertArrayEquals(new Person[]{
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45),
+            new Person("Николай", "Зимов", 30),
+            new Person("Иван", "Мельников", 20)
+        }, persons);
+    }
+
+    @Test
+    public void findFirstWithAge30UsingGuavaPredicate() {
+        List<Person> persons = Arrays.asList(getPersons());
+
+        Person person = FluentIterable
+            .from(persons)
+            .firstMatch(new Predicate<Person>() {
```

Comment:
As I understand exercise, here should not be anonymous class, you must declare class which implements Guava ```Predicate``` and use it's instance.

Reply:
Fixed